### ISSUE TITLE
iVim Pro Clone Repo isn’t working

### ISSUE THREAD

**Issue by ahayman** [terrychou/iVim#218]
I subscribed to iVim Pro in order to clone a repo from Github (a private one of mine).  Couldn’t get it to work:

- I was going to use an SSH, but I couldn’t change the “Cred Type” from “Username and Password”.  Tapping on the field does nothing.
- Trying to use my github username and password doesn’t work.  That’s not surprising as Github discontinued support for that authentication model: https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/

**Comment by terrychou** [comment]
1) the "Cred Type" switching problem

This is a UI issue started appearing in i(Pad)OS 15. I have fixed it in version 2.43 (it's also been released in the App Store), updating it will solve this problem.

2) use GitHub username & password

Yes, GitHub has stopped supporting to use username and password in third-party apps. However, you can create a personal access token (Settings > Developer settings > Personal access tokens) and use the token in place of the password.

Thanks for the reporting and sorry for the inconvenience.

**Comment by ahayman** [comment]
Updating fixed the "Cred Type" switching problem.

I still couldn't access my repo using ssh.  I tried both creating files for the pub/private keys, and manually pasting the data in.  Neither worked.  I tried the same keys on my Mac without issue.  Also, iVim doesn't seem to support the ssh url.  I tried both `git@github.com...` and `git+ssh://git@github.com...`, but both came up as unsupported.  When I tried to use the https url with the keys, iVim says "cloning..." but never finished.  I waited twenty minutes before closing it out.

Using a personal access token did work (and cloned immediately).  So I can access my repo now (yay!).  Using SSH keys seem to be broken though.
